include/itkBoxSpatialObjectToVTKPolyDataFilter.hxx

```diff
@@ -0,0 +1,142 @@
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkBoxSpatialObjectToVTKPolyDataFilter_hxx
+#define itkBoxSpatialObjectToVTKPolyDataFilter_hxx
+
+#include "itkBoxSpatialObjectToVTKPolyDataFilter.h"
+
+#include <vtkCubeSource.h>
+
+namespace itk
+{
+
+template< unsigned int VDimension >
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::BoxSpatialObjectToVTKPolyDataFilter()
+{
+  this->SetNumberOfRequiredInputs(1);
+  this->SetNumberOfRequiredOutputs(1);
+}
+
+
+template< unsigned int VDimension >
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::~BoxSpatialObjectToVTKPolyDataFilter()
+{
+}
+
+
+template< unsigned int VDimension >
+void
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::SetInput( const BoxSpatialObjectType * input )
+{
+  // Process object is not const-correct so the const_cast is required here
+  this->ProcessObject::SetNthInput( 0, const_cast< BoxSpatialObjectType * >( input ) );
+}
+
+
+template< unsigned int VDimension >
+const typename BoxSpatialObjectToVTKPolyDataFilter<VDimension>::BoxSpatialObjectType *
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::GetInput() const
+{
+  return itkDynamicCastInDebugMode< const BoxSpatialObjectType * >( this->GetPrimaryInput() );
+}
+
+
+template< unsigned int VDimension >
+const typename BoxSpatialObjectToVTKPolyDataFilter<VDimension>::BoxSpatialObjectType *
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::GetInput( unsigned int idx ) const
+{
+  return dynamic_cast< const BoxSpatialObjectType * >( this->ProcessObject::GetInput(idx) );
+}
+
+
+template< unsigned int VDimension >
+vtkPolyData *
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::GetOutput()
+{
+  return this->m_PolyData.GetPointer();
+}
+
+
+template< unsigned int VDimension >
+void
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::PrintSelf(std::ostream & os, Indent indent) const
+{
+  Superclass::Superclass::PrintSelf(os, indent);
+
+  os << indent << "PolyData output: ";
+    this->m_PolyData->Print( os );
+}
+
+
+template< unsigned int VDimension >
+void
+BoxSpatialObjectToVTKPolyDataFilter<VDimension>
+::Update()
+{
+  if( this->GetInput()->GetMTime() > this->m_ProcessedInputMTime )
+    {
+    BoxSpatialObjectType * input = const_cast< BoxSpatialObjectType * >( this->GetInput() );
+    input->Update();
```

Comment:
Hi,

SpatialObjects don't implement an "update" function.  It is essentially a no-op.    However, whenever you change a parameter of a spatialObject, (e.g., change the corners of the box) you are suppose to call "ComputeObjectBoundingBox" manually, because it is potentially an expensive operation.   ObjectBoundingBox exists in World coordinates, it is used by "IsInside", "ValueAt" and other functions that take-in points in World coordinates.  (FYI, the only things that exist in object coordinates are the object's parameters (e.g., radius for a sphere, corners of a box, etc).   Hope this helps.

Reply:
Fantastic explanation -- thanks! Will remove the `Update()` call.